docs/ibc/light-clients/setup.md

```diff
@@ -0,0 +1,110 @@
+
+```go
+var _ module.AppModuleBasic = AppModuleBasic{}
+
+// AppModuleBasic defines the basic application module used by the tendermint light client.
+// Only the RegisterInterfaces function needs to be implemented. All other function perform
+// a no-op.
+type AppModuleBasic struct{}
+
+// Name returns the tendermint module name.
+func (AppModuleBasic) Name() string {
+	return ModuleName
+}
+
+// RegisterLegacyAminoCodec performs a no-op. The Tendermint client does not support amino.
+func (AppModuleBasic) RegisterLegacyAminoCodec(*codec.LegacyAmino) {}
+
+// RegisterInterfaces registers module concrete types into protobuf Any. This allows core IBC
+// to unmarshal tendermint light client types.
+func (AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+	RegisterInterfaces(registry)
+}
+
+// DefaultGenesis performs a no-op. Genesis is not supported for the tendermint light client.
+func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
+	return nil
+}
+
+// ValidateGenesis performs a no-op. Genesis is not supported for the tendermint light cilent.
+func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config client.TxEncodingConfig, bz json.RawMessage) error {
+	return nil
+}
+
+// RegisterGRPCGatewayRoutes performs a no-op.
+func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {}
+
+// GetTxCmd performs a no-op. Please see the 02-client cli commands.
+func (AppModuleBasic) GetTxCmd() *cobra.Command {
+	return nil
+}
+
+// GetQueryCmd performs a no-op. Please see the 02-client cli commands.
+func (AppModuleBasic) GetQueryCmd() *cobra.Command {
+	return nil
+}
+```
+
+## Creating clients
+
+A client is created by executing a new `MsgCreateClient` transaction composed with a valid `ClientState` and initial `ConsensusState` encoded as protobuf `Any`s.
+
+Stateless checks are performed within the [`ValidateBasic`](https://github.com/cosmos/ibc-go/blob/7ae97694af/modules/core/02-client/types/msgs.go#L48) method of `MsgCreateClient`.
+
+```protobuf
+// MsgCreateClient defines a message to create an IBC client
+message MsgCreateClient {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // light client state
+  google.protobuf.Any client_state = 1 [(gogoproto.moretags) = "yaml:\"client_state\""];
+  // consensus state associated with the client that corresponds to a given
+  // height.
+  google.protobuf.Any consensus_state = 2 [(gogoproto.moretags) = "yaml:\"consensus_state\""];
+  // signer address
+  string signer = 3;
+}
+```
+
+Leveraging protobuf `Any` encoding allows core IBC to [unpack](https://github.com/cosmos/ibc-go/blob/7ae97694af/modules/core/keeper/msg_server.go#L28-L36) both the `ClientState` and `ConsensusState` into their respective interface types registered previously using the light client module's `RegisterInterfaces` method.
```

Comment:
should we link to a tag instead of commit?

Reply:
I was thinking we will probably need to address all these links to point to a v7 final release when its tagged. I've updated to use the `02-client-refactor-beta1` tag for now